File: editorial/chap09/cpp/answer_B62.cpp
#include <iostream>
#include <vector>
#include <string>
using namespace std;
 
// 입력
int N, M, A[100009], B[100009];
 
// 깊이 우선 탐색
vector<int> G[100009];
bool visited[100009];
 
// 깊이 우선 탐색의 흔적
vector<int> Path, Answer;
 
void dfs(int pos) {
	// 결승점에 도착한 경우
	if (pos == N) {
		Answer = Path;
		return;
	}
 
	// 그 외의 경우
	visited[pos] = true;
	for (int i = 0; i < G[pos].size(); i++) {
		int nex = G[pos][i];
		if (visited[nex] == false) {
			Path.push_back(nex); // 경로에 정점 nex 추가
			dfs(nex);
			Path.pop_back(); // 경로에서 정점 nex 제거
		}
	}
	return;
}
 
int main() {
	// 입력
	cin >> N >> M;
	for (int i = 1; i <= M; i++) {
		cin >> A[i] >> B[i];
		G[A[i]].push_back(B[i]);
		G[B[i]].push_back(A[i]);
	}
 
	// 깊이 우선 탐색
	for (int i = 1; i <= N; i++) visited[i] = false;
	Path.push_back(1); // 경로에 정점 1(시작점) 추가
	dfs(1);
 
	// 답 출력
	for (int i = 0; i < Answer.size(); i++) {
		if (i >= 1) cout << " ";
		cout << Answer[i];
	}
	cout << endl;
	return 0;
}

File: editorial/chap09/python/answer_B62.py
import sys
sys.setrecursionlimit(1 << 20)  # 재귀 상한이 기본적으로 1000으로 설정되어 있으므로 변경한다

# 입력
N, M = map(int, input().split())
g = [[] for _ in range(N)]
for i in range(M):
    A, B = map(int, input().split())
    A -= 1
    B -= 1
    g[A].append(B)
    g[B].append(A)

# 깊이 우선 탐색
visited = [False] * (N + 1)
path = []

def dfs(i: int):
    path.append(i)
    # 결승점에 도착한 경우
    if i == N - 1:
        # 답을 출력하고 종료
        for x in path:
            print(x + 1) # 1-indexed로 변경
        exit(0)

    # 그 외의 경우
    visited[i] = True
    for j in g[i]:
        if not visited[j]:
            dfs(j)
    path.pop()

dfs(0)
